app/models/mixins/miq_request_mixin.rb

```diff
@@ -185,9 +185,11 @@ def dialog_zone
     zone
   end
 
+  LOG_SERVERS_LIMIT = 10
   def mark_execution_servers
     options[:executed_on_servers] ||= []
     options[:executed_on_servers] << MiqServer.my_server.id
+    options[:executed_on_servers].shift if options[:executed_on_servers].size > LOG_SERVERS_LIMIT
```

Comment:
Is there value to keep the same server multiple times?

Reply:
I don't think so but Lucy should answer this.

Reply:
I think order is important

Reply:
This is from previous discussion: https://github.com/ManageIQ/manageiq/pull/17451#discussion_r196450889.

If we want to record all the servers in the order the tasks have been executed and limit the array size, maybe we can do this:
```
 options[:executed_on_servers] << MiqServer.my_server.id unless options[:executed_on_servers].last == MiqServer.my_server.id
```
cc @gmcculloug

Reply:
@lfu good suggestion, but it may not help if f there are more than one server which may process request.
 We could either sacrifice order and collect unique entries or sacrifice full history and limit size, I am not sure how we can do both.
It is a corner case, I am not sure what is the correct solution here and what limit should be in case we will decide to limit array size

Reply:
Sorry, I was really asking if an array of [1,1,1,1,1,1,1,1,1,1,1] was of any value?  If duplicates have no value but order of unique servers is important, a `Set` could be used to change:

`[1,2,1,2,1,2,1,2...]` to `[1,2]`... you can still limit how many are in the Set.

Reply:
The plus side of using a Set would be that you could retain "more" history as duplicate servers would not be included so you could go back further in time to get more unique servers listed.   Of course, this assumes duplicates aren't useful.

Reply:
I think order will be gone if we will use Set:
entry [1,2,1]  will be [1,2]

Reply:
> I think order will be gone if we will use Set:
> entry [1,2,1] will be [1,2]

Only if duplicates are have some value...  I have no idea, so I'm just asking.

Reply:
me too :) I am not sure what is correct solution here

Reply:
Review the comments from @lfu on the thread she listed above #17451 (comment):

> This array is to record the id of servers where the automate task has been processed. So all the logs from these servers would be collected in case of troubleshooting. The order of the ids in the array is not important in this case.

and

> If a long array is not a concern, we can get rid of calling uniq

So `uniq` still gives us the ability to see all the places it ran, which was the main purpose, and if we go this way I would skip limiting the array for now so we know where to collect logs from.  This is based on my assumption that most environments have a reasonable number of workers per-zone so the list would be contained by that.  If you feel this is incorrect than we should limit it.

This is a bit crazy, but since I worked on it I thought I would include it for fun.
If we want to keep a little sense of order the following logic clears out the current server record from the list and then appends it to the end and ensures we only return the `last` LOG_SERVERS_LIMIT elements.  Again, with removing dups we could probably drop this limit here as well.
```ruby
def last_test_servers(server_list, server_id)
  ((server_list -= [server_id]) << server_id).last(LOG_SERVERS_LIMIT)
end
```
or without limits:
```ruby
def last_test_servers(server_list, server_id)
  (server_list -= [server_id]) << server_id
end
```

Reply:
👍 It looks like good compromise